Add some docs for bqetl commands (#1846)
mozilla/bigquery-etl: docs/bqetl.md

# bqetl CLI

The `bqetl` command-line tool aims to simplify working with the bigquery-etl repository by supporting common workflows, such as creating, validating and scheduling queries or adding new UDFs.

## Installation

> Please follow the [Quick Start](https://github.com/mozilla/bigquery-etl#quick-start) for setting up bigquery-etl and the bqetl CLI.

Reviewer: ```suggestion
Follow the [Quick Start](https://github.com/mozilla/bigquery-etl#quick-start) to set up bigquery-etl and the bqetl CLI.
```

## Commands

To list all available commands in the bqetl CLI:

```bash
$ ./bqetl

> 
> Usage: bqetl [OPTIONS] COMMAND [ARGS]...
> 
>   CLI tools for working with bigquery-etl.
> 
> Options:
>   --version  Show the version and exit.
>   --help     Show this message and exit.
> 
> Commands:
>   alchemer    Commands for importing alchemer data.
>   dag         Commands for managing DAGs.
>   dependency  Build and use query dependency graphs.
>   dryrun      Dry run SQL.
>   format      Format SQL.
>   glam        Tools for GLAM ETL.
>   mozfun      Commands for managing mozfun routines.
>   query       Commands for managing queries.
>   routine     Commands for managing routines.
>   stripe      Commands for Stripe ETL.
>   view        Commands for managing views.

Reviewer: I would probably skip showing the output, as it's just one more thing to keep up to date.

```suggestion
```

Reviewer 2: This is a tough one. I agree there's danger of this getting out of date, but it's also a nice concise summary of what to expect, which could provide motivation for users to go ahead and invest a few minutes in getting the CLI installed to play around with it more.

Author: I think this is useful. Unfortunately, I couldn't find a way to get this information programmatically.

Reviewer: Could you run `./bqetl` by itself in CI (as you do in the example), capture its output, then paste it in here?

Author: I think this could be possible. My concern would be that the documentation process it might get harder to understand (some things happen in CI, some as Python scripts, some parts are static).

I'll open a follow-up issue to investigate this further at some point in the future.

```

See help for any command:

```bash
$ ./bqetl [command] --help
```

### `alchemer`

Commands for importing alchemer data in BigQuery.

#### `backfill`

```bash
# Import data from alchemer (surveygizmo) surveys into BigQuery.
# The date range is inclusive of the start and end values.
$ ./bqetl alchemer backfill --start-date=2021-01-01 --end-date=2021-02-01 --survey_id=xxxxxxxxxxx --api_token=xxxxxxxxxxxxxx --api_secret=xxxxxxxxxxxxxxx --destination_table=moz-fx-data-shared-prod.telemetry_derived.survey_gizmo_daily_attitudes
```

### `dag`

Commands for managing Airflow DAGs.

#### `create`

When creating new DAGs, the DAG name must have a `bqetl_` prefix. Created DAGs are added to the `dags.yaml` file.

```bash
# Create new DAG

> $ ./bqetl dag create bqetl_core --schedule-interval="0 2 * * *" --owner=example@mozilla.com --description="Tables derived from the legacy telemetry `core` ping sent by various mobile applications." --start-date=2019-07-25

Reviewer: Is there any way we could pull these examples from the source?

> # bqetl CLI

Reviewer: Can you reference this in the README, e.g. via a link? 

https://github.com/mozilla/bigquery-etl#the-bqetl-cli
